packages/content/src/content.ts

```diff
@@ -1,8 +1,292 @@
 /* Copyright (c) 2018 Environmental Systems Research Institute, Inc.
  * Apache-2.0 */
 
+import { ISpatialReference, IExtent } from "@esri/arcgis-rest-types";
+import { IItem, getItem } from "@esri/arcgis-rest-portal";
 import { request } from "@esri/arcgis-rest-request";
+import {
+  HubType, IHubContent, IHubGeography, IHubRequestOptions, IBBox,
+  getType, getCategory, getItemThumbnailUrl, getPortalApiUrl
+} from "@esri/hub-common";
 
+function buildUrl (base: string, path: string): string {
+  const baseUrl = base.endsWith('/') ? base : `${base}/`;
+  const url = new URL(path, baseUrl);
+  return url.toString();
+}
+
+// TODO: make this real; use request() under the hood?
+export function hubRequest (url: string, requestOptions?: IHubRequestOptions) {
+  return fetch(url, {
+    method: 'POST', // TODO: get from requestOptions?
+    headers: {
+      // TODO: base on request requestOptions?
+      'Content-Type': 'application/json'
+    },
+    // TODO: base on requestOptions.params?
+    // body: JSON.stringify(body)
+  }).then(resp => {
+    if (resp.ok) {
+      return resp.json();
+    } else {
+      throw resp.statusText;
+    }
+  });
+}
+
+function parseDatasetId(datasetId: string): { itemId: string, layerId: string } {
+  const [itemId, layerId] = datasetId ? datasetId.split('_') : [];
+  return { itemId, layerId }
+}
+
+function getContentFromHub(id: string, requestOptions?: IHubRequestOptions): Promise<IHubContent> {
+  const url = buildUrl(requestOptions && requestOptions.hubApiUrl, `/datasets/${id}`);
+  return hubRequest(url, requestOptions)
+  .then(resp => {
+    const dataset = resp && resp.data && resp.data[0];
+    const portalApiUrl = getPortalApiUrl(requestOptions && requestOptions.portalSelf);
+    return datasetToContent(dataset, portalApiUrl);
+  });
+}
+
+function getContentFromAgo(id: string, requestOptions?: IHubRequestOptions): Promise<IHubContent> {
```

Comment:
this method is mis-named since it would also apply to getting content from enterprise. Can we keep the name as `getContentFromPortal` @tomwayson